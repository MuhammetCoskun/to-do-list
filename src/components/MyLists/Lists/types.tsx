import { ToDoListItemType } from "../../ShowList/ToDoList/types";

export type ListType = {
  id: string;
  title: string;
  todos: ToDoListItemType[];
};

export type ListsProps = {
  lists: ListType[];
  onChooseList: (arg: string) => void;
};
