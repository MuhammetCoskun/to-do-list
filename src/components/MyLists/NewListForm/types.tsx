export type NewListFormProps = {
  onAddList: (listName: string) => void;
};
