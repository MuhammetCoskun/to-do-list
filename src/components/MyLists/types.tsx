import { ListType } from "./Lists/types";

export type MyListsProps = {
  onChooseList: (arg: string) => void;
  onAddList: (title: string) => void;
  lists: ListType[];
};
