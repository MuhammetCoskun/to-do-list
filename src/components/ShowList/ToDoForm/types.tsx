export interface ToDoFormProps {
  onAddTodo: (todo: string) => void;
  onRemoveTodo: () => void;
  onRemoveList: () => void;
}
