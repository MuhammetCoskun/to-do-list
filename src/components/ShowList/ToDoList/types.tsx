import { ListType } from "../../MyLists/Lists/types";

export type ToDoListItemType = {
  id: string;
  text: string;
  createdAt: string;
  isDone: boolean;
  doneAt?: string;
};

export type ToDoListItemProps = {
  id: string;
  key?: string;
  text: string;
  onRemove: (id: string) => void;
};

export type ToDoListProps = {
  list: ListType;
  onRemove: (id: string) => void;
};
