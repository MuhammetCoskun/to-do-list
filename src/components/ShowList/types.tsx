import { ListType } from "../MyLists/Lists/types";

export type ShowListProps = {
  listToShow: ListType;
  onAddTodo: (text: string) => void;
  onRemoveTodo: (id: string) => void;
};
