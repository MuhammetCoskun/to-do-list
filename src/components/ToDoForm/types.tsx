export interface TodoFormProps {
  onAddTodo: (todo: string) => void;
}
