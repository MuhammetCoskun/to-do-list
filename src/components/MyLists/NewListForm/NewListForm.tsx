import React, { useRef, useState } from "react";
import { NewListFormProps } from "./types";

const NewListForm: React.FC<NewListFormProps> = ({ onAddList }) => {
  const newListRef = useRef<HTMLInputElement>(null);
  const [inputIsValid, setInputIsValid] = useState<boolean>(true);

  const submitHandler = (e: React.FormEvent) => {
    e.preventDefault();
    const newList = newListRef.current!.value;
    if (newList.trim().length === 0) {
      setInputIsValid(false);
      return;
    }
    onAddList(newList);
    newListRef.current!.value = "";
    setInputIsValid(true);
  };

  return (
    <>
      <form onSubmit={submitHandler}>
        <input
          type="text"
          placeholder="Enter a new List"
          aria-label="new list name"
          ref={newListRef}
          data-testid="input"
          className="new list"
        />
        <button
          aria-label="create new list"
          data-testid="button"
          className="btn create"
        >
          +
        </button>
      </form>
      {!inputIsValid && (
        <p data-testid="paragraph">List name can not be empty..</p>
      )}
    </>
  );
};

export default NewListForm;
