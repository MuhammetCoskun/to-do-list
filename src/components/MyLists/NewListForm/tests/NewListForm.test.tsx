import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import NewListForm from "../index";
import { NewListFormProps } from "../types";

const mockOnAddList = jest.fn();
const customRender = (props: NewListFormProps) => {
  render(<NewListForm {...props} />);
};
describe("NewListForm component", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<NewListForm onAddList={mockOnAddList} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders error message on invalid input", () => {
    customRender({ onAddList: mockOnAddList });
    const buttonElement = screen.getByRole("button");
    userEvent.click(buttonElement);
    expect(
      screen.getByText("List name can not be empty..")
    ).toBeInTheDocument();
  });
  test("when button clicked it should call onAddList callback", () => {
    customRender({ onAddList: mockOnAddList });
    const inputElement = screen.getByTestId("input");
    const buttonElement = screen.getByTestId("button");
    userEvent.type(inputElement, "New List!");
    userEvent.click(buttonElement);
    expect(mockOnAddList).toHaveBeenCalledWith("New List!");
  });
});
