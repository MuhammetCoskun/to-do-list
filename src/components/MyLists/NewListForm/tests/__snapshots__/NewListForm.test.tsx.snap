// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewListForm component renders correctly 1`] = `
<DocumentFragment>
  <form>
    <input
      aria-label="new list name"
      class="new list"
      data-testid="input"
      placeholder="Enter a new List"
      type="text"
    />
    <button
      aria-label="create new list"
      class="btn create"
      data-testid="button"
    >
      +
    </button>
  </form>
</DocumentFragment>
`;
